app/src/main/java/com/inveitix/android/clue/database/FireBaseLoader.java

```diff
@@ -44,23 +44,10 @@ public void onDataChange(DataSnapshot dataSnapshot) {
 
                     Museum museum = postMuseum.getValue(Museum.class);
 
-                    if (museums.size() < 1) {
-                        museums.add(museum);
-                        Log.e("ADD", "ÄDDed 1");
-                    } else {
-                        for (int i = 0; i < museums.size(); i++) {
-                            if (museums.get(i).getId() == museum.getId()) {
-                                museums.remove(i);
-                                museums.add(museum);
-
-                                Log.e("ADD", "ÄDDed 2");
-                            } else {
-                                museums.add(museum);
-                                Log.e("ADD", "ÄDDed 3");
-                            }
-                        }
-                    }
+                    museums.add(museum);
                 }
+
+                duplicateCheck(museums);
```

Comment:
Do a check before adding a museum to the list, if the museum is already in the list, not after that.